wrapanapi/systems/rhevm.py

```diff
@@ -881,26 +909,58 @@ def get_template(self, name=None, uuid=None):
             )
         return matches[0]
 
-    def create_template(self, template_name, vm_name, cluster_name=None):
+    def create_template(self, template_name, vm_name, cluster_name=None, storage_domain_name=None):
         """
         Create a template based on a VM.
 
         Creates on the same cluster as the VM unless 'cluster_name' is specified
+
+        http://ovirt.github.io/ovirt-engine-sdk/master/services.m.html
+            #ovirtsdk4.services.TemplatesService.add
         """
         vm = self.get_vm(vm_name)
+        vm.refresh(follow='disk_attachments')  # include disk_attachment refs
 
         if cluster_name:
             cluster = self.get_cluster(cluster_name)
         else:
             cluster = vm.cluster
 
-        new_template = types.Template(name=template_name, vm=vm.raw, cluster=cluster)
+        template_kwargs = dict(
+            name=template_name,
+            vm=vm.raw,
+            cluster=cluster,
+        )
+
+        if storage_domain_name:
+            template_kwargs.update({'storage_domain': self.get_storage_domain(storage_domain_name)})
+        # FIXME: pick domain from the VM's disk storage domains
+        # might not need to pass explicitly in this case anyway
+        # ovirt API a bit complicated here, failing to pickup on the setting
+        # It needs a rich VM object, with its disk attachment assignments filled out with disk IDs
+        # and the domain set for each disk, or globally for the template
+        # else:
+        #     domains = vm.storage_domains
+        #     if len(set([d.name for d in domains])) > 1:
+        #         self.logger.warning(
+        #             'More than one storage domain for VM disks,'
+        #             'picking one for creating template: %r',
+        #             domains
+        #         )
+        #     storage_domain = domains[0]
+
+        new_template = types.Template(**template_kwargs)
+
         self.api.system_service().templates_service().add(new_template)
 
         # First it has to appear
         wait_for(
-            lambda: self.does_template_exist(template_name),
-            num_sec=30 * 60, message="template exists", delay=45)
+            func=self.does_template_exist,
+            func_args=[template_name],
+            num_sec=300,
+            message="template exists",
+            delay=5
```

Comment:
Same question about the delay here. And also about the decrease in `num_sec`.

Reply:
I think its overkill to delay for 45 seconds between each check for the template to show up. 30 minutes is also an incredibly long time to wait for the template record to show up.  There is another call below to check that the template is in 'OK' status.

300 might be too short though, maybe 600 and add it to the function kwargs so it can be passed in for a "slow" provider.